client/web/src/enterprise/codeintel/configuration/CodeIntelConfigurationPage.tsx

```diff
@@ -0,0 +1,610 @@
+        [setSelectedTab]
+    )
+
+    return (
+        <div className="overflow-auto mb-2">
+            <ul className="nav nav-tabs d-inline-flex d-sm-flex flex-nowrap text-nowrap">
+                <li className="nav-item">
+                    {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                    <a
+                        href=""
+                        onClick={onGlobalPolicies}
+                        className={classNames('nav-link', selectedTab === 'globalPolicies' && 'active')}
+                        role="button"
+                    >
+                        <span className="text-content" data-tab-content="Global policies">
+                            Global policies
+                        </span>
+                    </a>
+                </li>
+                <li className="nav-item">
+                    {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                    <a
+                        href=""
+                        onClick={onRepositoryPolicies}
+                        className={classNames('nav-link', selectedTab === 'repositoryPolicies' && 'active')}
+                        role="button"
+                    >
+                        <span className="text-content" data-tab-content="Repository-specific policies">
+                            Repository-specific policies
+                        </span>
+                    </a>
+                </li>
+                <li className="nav-item">
+                    {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                    <a
+                        href=""
+                        onClick={onIndexConfiguratino}
+                        className={classNames('nav-link', selectedTab === 'indexConfiguration' && 'active')}
+                        role="button"
+                    >
+                        <span className="text-content" data-tab-content="Index configuration">
+                            Index configuration
+                        </span>
+                    </a>
+                </li>
+            </ul>
+        </div>
+    )
+}
+
+interface PoliciesListProps {
+    policies?: CodeIntelligenceConfigurationPolicyFields[]
+    deletePolicy?: (id: string, name: string) => Promise<void>
+    disabled: boolean
+    indexingEnabled: boolean
+    buttonFragment?: JSX.Element
+    history: H.History
+}
+
+const PoliciesList: FunctionComponent<PoliciesListProps> = ({ policies, buttonFragment, ...props }) =>
+    policies === undefined ? (
+        <LoadingSpinner className="icon-inline" />
+    ) : (
+        <>
+            {policies.length === 0 ? (
+                <div>No policies have been defined.</div>
+            ) : (
+                <CodeIntelligencePolicyTable {...props} policies={policies} />
+            )}
+            {buttonFragment}
+        </>
+    )
+
+const describeRetentionPolicy: (policy: CodeIntelligenceConfigurationPolicyFields) => JSX.Element = policy =>
```

Comment:
Essentially it's React component. Is there a reason to define/name it in a different manner? 

```suggestion
const DescribeRetentionPolicy: FunctionComponent<{ policy: CodeIntelligenceConfigurationPolicyFields }> = ({ policy }) =>
```